api/src/main/java/org/openmrs/module/fhir2/api/translators/EncounterPeriodTranslator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators;
+
+import javax.annotation.Nonnull;
+
+import java.util.Date;
+
+import org.hl7.fhir.r4.model.Period;
+
+public interface EncounterPeriodTranslator extends ToFhirTranslator<Date, Period>, OpenmrsFhirTranslator<Date, Period> {
```

Comment:
Wouldn't it be better to structure this as an interface that translates between an OpenMRS type and a date? That way, we only need a single interface (`EncounterPeriodTranslator`) that has two specialisations (one for `Encounter`, one for `Visit`) and we don't have to do weird things like returning `Pairs`